robowflex_library/include/robowflex_library/robot.h

```diff
@@ -450,11 +461,27 @@ namespace robowflex
              *  \param[in] pose Pose of the \a region.
              *  \param[in] orientation Mean orientation.
              *  \param[in] tolerances Tolerance about \a orientation.
+             *  \param[in] scene Optional scene to do collision checking against.
+             *  \param[in] verbose If true, will give verbose collision checking output.
              */
-            IKQuery(const std::string &group, const GeometryConstPtr &region, const RobotPose &pose,
-                    const Eigen::Quaterniond &orientation,
-                    const Eigen::Vector3d &tolerance = constants::ik_rot_tolerance,
-                    const ScenePtr &scene = nullptr, bool verbose = false);
+            IKQuery(const std::string &group,                                        //
+                    const GeometryConstPtr &region,                                  //
+                    const RobotPose &pose,                                           //
+                    const Eigen::Quaterniond &orientation,                           //
+                    const Eigen::Vector3d &tolerance = constants::ik_rot_tolerance,  //
+                    const ScenePtr &scene = nullptr,                                 //
+                    bool verbose = false);
+
+            /** Constructor. Initialize a basic IK query to grasp an object.
+             *  \param[in] group Group to set.
+             *  \param[in] offset Offset of end-effector from object frame.
+             *  \param[in] scene Scene that object is in.
+             *  \param[in] object Name of object to grasp.
+             */
+            IKQuery(const std::string &group,  //
+                    const RobotPose &offset,   //
+                    const ScenePtr &scene,     //
+                    const std::string &object);
```

Comment:
Would it be interesting to have input tolerances here as well?

Reply:
I believe in the dataset project you have a box with XYZ tolerances? That could be easily integrated here as well.